File: src/statusbar/modules/pane-info.ts
import type {
  StatusBarModule,
  StatusBarModuleContext,
  Segment,
} from "../types.ts";

export const paneInfoModule: StatusBarModule = {
  id: "pane-info",
  render(ctx: StatusBarModuleContext): Segment[] {
    const icon = ctx.icons ? " " : "";
    const count = ctx.metrics.paneCount;
    const title = ctx.metrics.paneTitle;

    let text: string;
    if (title && title !== ctx.session.name) {
      text = ` ${icon}${count}P ${title} `;
    } else {
      text = ` ${icon}${count}P `;
    }

    return [
      {
        text,
        fg: ctx.colors.fg,
        bg: ctx.colors.bg,
      },
    ];
  },
};

File: rust/crates/maxmux-statusbar/src/modules/pane_info.rs
use crate::modules::{ModuleContext, StatusBarModule};
use crate::types::Segment;

pub struct PaneInfoModule;

impl StatusBarModule for PaneInfoModule {
    fn id(&self) -> &str {
        "pane_info"
    }

    fn render(&self, ctx: &ModuleContext) -> Vec<Segment> {
        let count = ctx.metrics.pane_count;

        // Show title only if different from session name.
        let show_title = ctx
            .metrics
            .pane_title
            .as_ref()
            .is_some_and(|t| t != &ctx.session.name);

        let text = if ctx.icons {
            if show_title {
                format!(
                    " \u{f0135} {}P {} ",
                    count,
                    ctx.metrics.pane_title.as_deref().unwrap_or("")
                )
            } else {
                format!(" \u{f0135} {}P ", count)
            }
        } else if show_title {
            format!(
                " {}P {} ",
                count,
                ctx.metrics.pane_title.as_deref().unwrap_or("")
            )
        } else {
            format!(" {}P ", count)
        };

        vec![Segment::new(&text, &ctx.colors.fg, &ctx.colors.bg)]
    }
}
